### ISSUE TITLE
RuntimeError: Expected object of type torch.FloatTensor but found type torch.cuda.FloatTensor for argument #4 'mat1'

### ISSUE THREAD

**Issue by Jeffrey28** [p-christ/Deep-Reinforcement-Learning-Algorithms-with-PyTorch#31]
I run the Cart_Pole.py and got the error.

**Comment by Jeffrey28** [comment]
And if I choose “use_GPU" is True. It works well. However, when it ran with A3C, I got the error: Runtime Error: CUDA error (3): initialization error. So, do you have any suggestions for me?

**Comment by lml519** [comment]
yeah,i have the same problem, when i run python3 results/Cartpole.py

**Comment by p-christ** [comment]
GPU doesn't work with A3C because A3C uses multiprocessing - try turning off the GPU to see if it works.  For small games anyway it is faster to run it with the GPU turned off.

**Comment by lml519** [comment]
@p-christ  but when i turn off the gpu. it will display 
`RuntimeError: Expected object of type torch.FloatTensor but found type torch.cuda.FloatTensor for argument #4 'mat1'`

**Comment by Jeffrey28** [comment]
yeah

> @p-christ but when i turn off the gpu. it will display
> `RuntimeError: Expected object of type torch.FloatTensor but found type torch.cuda.FloatTensor for argument #4 'mat1'`

yeah, the same problem.

**Comment by p-christ** [comment]
Have you also set: config.use_GPU = False   ?

**Comment by Jeffrey28** [comment]
> Have you also set: config.use_GPU = False ?

Yeah, I have set.

**Comment by ztz695183179** [comment]
Have you solved this problem yet?I have the same problem when I run results/Cartpole with any agent.

**Comment by ai-xiaodi** [comment]
I also have the same problem when I run the cartpole

**Comment by ai-xiaodi** [comment]
I have the problem “RuntimeError: Expected object of device type cuda but got device type cpu for argument #1 'self' in call to _th_addmm”，any suggestions for me?
